tools/wake-format/emitter.cpp

```diff
@@ -27,38 +27,19 @@
 
 #define WALK(func) [this](ctx_t ctx, CSTElement node) { return func(ctx, node); }
 
-bool fits(const wcl::doc_builder& bdr, ctx_t ctx, wcl::doc doc) {
-  if (bdr.has_newline()) {
-    return bdr.last_width() + doc.first_width() <= Emitter::max_column_width;
-  } else {
-    return bdr.last_width() + doc.first_width() + ctx.width <= Emitter::max_column_width;
-  }
-}
-
-bool requires_nl(ctx_t ctx, CSTElement node) {
-  return node.id() == CST_BLOCK || node.id() == CST_REQUIRE;
-}
+static bool requires_nl(uint8_t type) { return type == CST_BLOCK || type == CST_REQUIRE; }
 
 static bool is_expression(uint8_t type) {
-  return type == CST_ID || type == CST_APP || type == CST_LITERAL;
+  return type == CST_ID || type == CST_APP || type == CST_LITERAL || type == CST_HOLE ||
+         type == CST_BINARY;
 }
 
-void Emitter::walk_rhs(wcl::doc_builder& bdr, ctx_t ctx, CSTElement& node) {
-  auto nested_fmt = fmt().nest(fmt().walk(WALK(walk_node)));
-  if (requires_nl(ctx, node)) {
-    bdr.append(nested_fmt.compose(ctx, node));
-    return;
-  }
-
+decltype(auto) Emitter::rhs_fmt() {
```

Comment:
Actually fix this: I learned this should be `auto` instead. When its `decltype(auto)` we open ourselves up to some issues.